qtconsole/frontend_widget.py

```diff
@@ -138,9 +134,6 @@ class FrontendWidget(HistoryConsoleWidget, BaseFrontendMixin):
     lexer_class = DottedObjectName(config=True,
         help="The pygments lexer class to use."
     )
-    is_complete_timeout = Float(0.25, config=True,
-        help="Seconds to wait for is_complete replies from the kernel."
```

Comment:
Instead of a timeout, I added the fallback code to the event handler for user events. I think this is better than a timeout.

With a timeout, we would need to have to think what is happening if the user continued editing in the meantime. Simply falling back once the user continues editing mitigates that problem.

I think it's also what we actually want: if the user is sitting in front of the computer waiting for the result of the execution, s/he won't continue typing. Continued editing is a good hint that the line is not complete.